File: app/src/main/java/com/mancode/financetracker/database/viewmodel/AccountViewModel.java
package com.mancode.financetracker.database.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.mancode.financetracker.database.entity.AccountEntity;
import com.mancode.financetracker.database.views.AccountExtended;
import com.mancode.financetracker.repository.DataRepository;

import java.util.List;

/**
 * Created by Manveru on 02.02.2018.
 */

public class AccountViewModel extends AndroidViewModel {

    private DataRepository mRepository;
    private LiveData<List<AccountExtended>> mAllAccounts;

    public AccountViewModel(@NonNull Application application) {
        super(application);
        mRepository = DataRepository.Companion.getInstance(application);
        mAllAccounts = mRepository.getAllAccounts();
    }

    public LiveData<List<AccountExtended>> getAllAccounts() {
        return mAllAccounts;
    }

    public void insert(AccountEntity account) {
        mRepository.insertAccount(account);
    }
}

File: app/src/main/java/com/mancode/financetracker/viewmodel/AccountViewModel.kt
package com.mancode.financetracker.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.mancode.financetracker.database.entity.AccountEntity
import com.mancode.financetracker.database.views.AccountExtended
import com.mancode.financetracker.repository.AccountsRepository
import com.mancode.financetracker.utils.InjectorUtils

/**
 * Created by Manveru on 02.02.2018.
 */
class AccountViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: AccountsRepository = InjectorUtils.getAccountsRepository(application)

    val allAccounts: LiveData<List<AccountExtended>> = repository.allAccountsExt

    fun insert(account: AccountEntity) {
        repository.insert(account)
    }
}